File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> nums_set(nums.begin(), nums.end());

        int longest = 0;
        for (int num: nums_set) {
            if (nums_set.count(num - 1)) { 
                continue;
            }

            int length = 1;

            while (nums_set.count(num + length)) {
                length++;
            }

            longest = max(longest, length);
        }

        return longest;
    }
};

File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        n, max_n = 0, 0

        if nums[0] == 1:
            n, max_n = 1, 1
        
        for i in range(1, len(nums)):
            if nums[i] == 0:
                max_n = max(n, max_n)
                n = 0
            else:
                n += 1
        
        return max(n, max_n)